src/signal-handling.c

```diff
@@ -129,6 +129,35 @@ static size_t jl_safe_read_mem(const volatile char *ptr, char *out, size_t len)
     return i;
 }
 
+static double profile_autostop_time = -1.0;
+static double profile_peek_duration = 1.0; // seconds
+
+double jl_get_profile_peek_duration(void) {
```

Comment:
style nit: functions start with `{` on the next line
```suggestion
double jl_get_profile_peek_duration(void)
{
```